cultivation/scripts/flashcore/cli/main.py

```
@@ -0,0 +1,223 @@
+            _display_overall_stats(console, stats_data)
+
+            if not stats_data['total_cards']:
+                console.print("[yellow]No cards found in the database.[/yellow]")
+                return
+
+            _display_deck_stats(console, stats_data)
+            _display_state_stats(console, stats_data)
+
+    except FlashcardDatabaseError as e:
+        console.print(f"[bold]A database error occurred: {e}[/bold]")
+        raise typer.Exit(code=1)
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)